service/history/replicationTaskFetcher.go

```diff
@@ -208,86 +225,51 @@ func (f *ReplicationTaskFetcherImpl) fetchTasks() {
 		f.config.ReplicationTaskFetcherAggregationInterval(),
 		f.config.ReplicationTaskFetcherTimerJitterCoefficient(),
 	))
-
-	requestByShard := make(map[int32]*request)
```

Comment:
I'm not sure I understand why you promoted this to be an instance level field rather than local variable.  There might be concurrency related concerns with this change.  Can you provide some explanation as a comment?

Reply:
> There might be concurrency related concerns with this change. 

The variable is within an event loop, there is no concurrent change.
Making this a instance variable will enable better testing, which will be after the kafka deprecation